### PR TITLE
FFT

### PR DESCRIPTION
**Pull request by laanwj**
Info from `src/lib/drivers/include/fft.h` `lib/drivers/fft.c`

### CODE REVIEW

File: k210.svd [hunk 1]
```diff
@@ -998,11 +998,223 @@
...
+                                </enumeratedValue>
+                            </enumeratedValues>
+                        </field>
+                        <field>
+                            <name>mode</name>
+                            <description>FFT mode</description>
+                            <bitRange>[3:3]</bitRange>
+                            <enumeratedValues>
+                                <enumeratedValue>
+                                    <name>backward</name>
```
Comment (Disasm) [inline k210.svd]: According to [this](https://www.gnu.org/software/gsl/doc/html/fft.html), *forward FFT* is `FFT` and *backward FFT* is a variant of *inverse FFT*, `IFFT`. This is not consistent with the information from `fft.h` and `fft.c`.
I suggest naming these variants `fft` and `ifft`.
```suggestion
                                    <name>fft</name>
```

File: k210.svd [hunk 2]
```diff
@@ -998,11 +998,223 @@
...
+                                    <description>FFT mode</description>
+                                    <value>0</value>
+                                </enumeratedValue>
+                                <enumeratedValue>
+                                    <name>forward</name>
```
Comment (Disasm) [inline k210.svd]:
```suggestion
                                    <name>ifft</name>
```

File: k210.svd [hunk 3]
```diff
@@ -998,11 +998,223 @@
...
+                    <name>input_fifo</name>
+                    <description>FFT input data fifo</description>
                     <addressOffset>0x00</addressOffset>
+                    <size>64</size>
+                    <fields>
+                        <field>
+                            <name>input_fifo</name>
+                            <description>Input fifo</description>
+                            <bitRange>[63:0]</bitRange>
+                        </field>
```
Comment (Disasm) [inline k210.svd]: I suggest removing this field because it does not provide any useful interface to the register.
Same for `output_fifo`.

**Review by Disasm — CHANGES_REQUESTED** [review]
Thanks!

File: k210.svd (hunk 1 above)
Comment (laanwj) [reply k210.svd]: I was really confused here too, so I used the naming from the enums that are used in the  example code, even though the comment seems to conflict. If you're sure about ifft/fft that's better naming, though.

File: k210.svd (hunk 3 above)
Comment (laanwj) [reply k210.svd]: OK

**Pull request merged**